src/App.jsx

```diff
@@ -1,17 +1,47 @@
 import { hot } from 'react-hot-loader';
 import { Component } from 'react';
 import { BrowserRouter, Switch, Route } from 'react-router-dom';
+import { ApolloProvider } from 'react-apollo';
+import { ApolloClient } from 'apollo-client';
+import { HttpLink } from 'apollo-link-http';
+import { RetryLink } from 'apollo-link-retry';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import { CachePersistor } from 'apollo-cache-persist';
+import storage from 'localforage';
 import Projects from './views/Projects';
+import Project from './views/Project';
 
 @hot(module)
 export default class App extends Component {
+  cache = new InMemoryCache();
+  persistence = new CachePersistor({
+    cache: this.cache,
+    storage,
+  });
+  link = new RetryLink().split(
+    operation => operation.getContext().link === 'github',
+    new HttpLink({
+      uri: 'https://api.github.com/graphql',
+      headers: {
+        authorization: 'Bearer 2365e5be2b67c4a07335ceafd64273211995c7a2',
```

Comment:
We will want to pull this value from the environment instead of directly injecting it into the app. You will want to make sure this token is revoked now since it is public and could potentially be cause to perform undesired operations on your behalf.

Reply:
ok! yeah i removed it actually, and the key is automatically unactivated once it is commited,
i try using cherry pick, but i think i hadn't done it carefully thus it appears here again